src/lib/meter/meter-group/meter-group.ts

```diff
@@ -0,0 +1,188 @@
+
+  /**
+   * The shape of each meter in the group.
+   * @default 'default'
+   * @attribute inner-shape
+   */
+  @property({ reflect: true, attribute: 'inner-shape' }) public innerShape: MeterInnerShape = 'default';
+
+  public get labels(): NodeList {
+    return this._internals.labels;
+  }
+
+  public get form(): HTMLFormElement | null {
+    return this._internals.form;
+  }
+
+  @state() private _hasSlottedHeadingContent = false;
+
+  @queryAssignedNodes() private _labelNodes: Node[];
+  @queryAssignedNodes({ slot: 'value' }) private _valueNodes: Node[];
+  @queryAssignedElements({ selector: 'forge-meter' }) private _meters: IMeterComponent[];
+
+  private _internals: ElementInternals;
+
+  constructor() {
+    super();
+    this._internals = this.attachInternals();
+  }
+
+  public connectedCallback(): void {
+    super.connectedCallback();
+    setDefaultAria(this, this._internals, { role: 'group' });
+  }
+
+  public willUpdate(changedProperties: PropertyValues<this>): void {
+    const keys = Array.from(changedProperties.keys());
+    if (keys.some(key => ['direction', 'min', 'max'].includes(key.toString()))) {
+      this._syncMeters();
+    }
+  }
+
+  public render(): TemplateResult {
+    return html`
+      <div part="root" class="forge-meter-group" @change=${this._debounceMeterChange}>
+        <div class=${classMap({ heading: true, 'not-empty': this._hasSlottedHeadingContent })} @slotchange=${this._handleHeadingSlotChange}>
+          <div class="label"><slot name="label"></slot></div>
+          <div class="value"><slot name="value"></slot></div>
+        </div>
+        <div part="track" class=${classMap({ track: true, tickmarks: this.tickmarks })}>
+          <slot @slotchange=${this._debounceMeterChange}></slot>
+        </div>
+      </div>
+    `;
+  }
+
+  /**
+   * Applies the min and max values to each meter in the group. These should always be set on the
+   * meter group component to ensure that all meters are in sync.
+   */
+  private _syncMeters(): void {
+    this._meters.forEach(meter => {
+      meter.direction = this.direction;
+      meter.min = this.min;
+      meter.max = this.max;
+    });
+  }
+
+  /**
+   * Debounce the handler to reduce the number of times it's called during rapid changes,
+   * especially when the component is first connected.
+   */
+  private _debounceMeterChange = debounce(this._handleMeterChange.bind(this), 0);
+
+  /**
+   * Handles changes to the meters in the group, updating their arrangement relative to each other.
+   */
+  private _handleMeterChange(evt: Event): void {
+    evt.stopPropagation();
+
+    // Iterate through each slotted meter, applying z-indices based on their order and setting an
+    // inset equal to the widths of all preceding meters.
+    this._meters.reduce((inset, meter, index, meters) => {
+      meter.style.setProperty('--z-index', `${meters.length - index}`);
+      meter.style.setProperty('--inset', `${inset}%`);
```

Comment:
Should we use prefixed and private named CSS vars here?

Reply:
I think so but the linter was complaining about it